src/ScriptBuilder/CSharpCounterWriter.cs

```diff
@@ -0,0 +1,103 @@
+﻿namespace NServiceBus.Metrics.PerformanceCounters
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+
+    static class CSharpCounterWriter
+    {
+        public static void WriteCode(string scriptPath, IEnumerable<TimerDefinition> timers, IEnumerable<MeterDefinition> meters, Dictionary<string, string> legacyInstanceNameMap)
```

Comment:
Yes we do actively. The approved file is a CSharp file that gets included in the SetupFixture of the tests. Therefore we not only approve the structure of the file but also compile and use it.